lib/json_schemer/schema/base.rb

```diff
@@ -615,16 +615,15 @@ def resolve_ids(schema, ids = {}, parent_uri = nil, pointer = '')
         if schema.is_a?(Array)
           schema.each_with_index { |subschema, index| resolve_ids(subschema, ids, parent_uri, "#{pointer}/#{index}") }
         elsif schema.is_a?(Hash)
-          id = schema[id_keyword]
-          uri = join_uri(parent_uri, id)
-          unless uri == parent_uri
-            ids[uri.to_s] = {
-              schema: schema,
-              pointer: pointer
-            }
-          end
-          if definitions = schema['definitions']
-            definitions.each { |key, subschema| resolve_ids(subschema, ids, uri, "#{pointer}/definitions/#{key}") }
+          uri = join_uri(parent_uri, schema[id_keyword])
+          schema.each do |key, value| 
+            if key == id_keyword &&  uri != parent_uri
+                ids[uri.to_s] = {
```

Comment:
Really minor, but this has some extra indentation now

Reply:
updated